src/lib.rs

```diff
@@ -150,6 +166,61 @@ pub mod error;
 mod session;
 use crate::session::{Cmd, Session};
 
+/// Builder pattern for Client
+#[derive(Default, Clone, Debug)]
+pub struct ClientBuilder<C>
+where
+    C: connect::Connect + Send + Sync + Clone + Unpin,
+{
+    capabilities: Option<webdriver::capabilities::Capabilities>,
+    connector: C,
+}
+
+#[cfg(feature = "rustls-tls")]
+impl ClientBuilder<hyper_rustls::HttpsConnector<hyper::client::HttpConnector>> {
+    /// Create a builder with Rustls connector, available with feature `rustls-tls`
+    pub fn rustls() -> Self {
+        Self::new(hyper_rustls::HttpsConnector::with_native_roots())
+    }
+}
+
+#[cfg(feature = "openssl-tls")]
+impl ClientBuilder<hyper_tls::HttpsConnector<hyper::client::HttpConnector>> {
+    /// Create a builder with OpenSSL connector, available with feature `openssl-tls`
+    pub fn openssl() -> Self {
+        Self::new(hyper_tls::HttpsConnector::new())
+    }
+}
+impl<C> ClientBuilder<C>
+where
+    C: connect::Connect + Send + Sync + Clone + Unpin + 'static,
+{
+    /// Create a new builder with TLS connector
+    /// # Arguments
+    /// * `connector` - TLS connector for [`Client`][hyper::client::Client]
+    pub fn new(connector: C) -> Self {
+        Self {
+            capabilities: None,
+            connector,
+        }
+    }
+    /// Set chosen webdriver capabilities
```

Comment:
Not sure what you mean by "chosen" here? Can this just be
```rust
/// Pass the given WebDriver capabilities to the browser.
```

Reply:
Also, in Rust, the convention is generally to leave an empty line between functions. So a blank line before this line.